index.js

```diff
@@ -3,48 +3,59 @@ const inputTitle = document.querySelector('#title');
 const inputAuthor = document.querySelector('#author');
 const addBtn = document.querySelector('#add');
 
-const savedData = localStorage.getItem('savedInput');
-
-let collection = [];
+class Book {
+  constructor(savedData = []) {
+    this.arr = savedData;
+  }
 
-if (savedData && savedData !== null) {
-  collection = JSON.parse(savedData);
-}
+  saveData(data) {
+    let existing = JSON.parse(localStorage.getItem('book'));
+    existing = existing || [];
+    this.arr = existing;
+    this.arr.push(data);
+    localStorage.setItem('book', JSON.stringify(this.arr));
+  }
 
-const displayData = () => {
-  divbooks.innerHTML = '';
-  collection.forEach((value, index) => {
-    divbooks.innerHTML += `
-            <div class="books">
-            <ul>
-                <li class="title">${value.name}</li>
-                <li class="author">${value.author}</li>
-            </ul>
-             <button id="remove" onclick="removeBook(${index});">remove</button>
-             <hr>
-           </div>`;
-  });
-};
+  removeBook = (index) => {
+    if (index !== null && index !== undefined) {
+      this.arr.splice(index, 1);
+      localStorage.setItem('book', JSON.stringify(this.arr));
+      this.getData();
+    }
+  };
 
-displayData();
-const saveData = () => localStorage.setItem('savedInput', JSON.stringify(collection));
-const removeBook = (index) => {
-  if (index !== null && index !== undefined) {
-    collection.splice(index, 1);
-    saveData();
-    displayData();
+  getData() {
+    divbooks.innerHTML = '';
+    this.arr.forEach((value, index) => {
+      divbooks.innerHTML += `
+              <div class="books">
+              <div class="list-btn">
+              <ul class="list">
+                  <li class="title">${value.name}</li>
+                  <p class="by">by</p>
+                  <li class="author">${value.author}</li>
+              </ul>
+              <button id="remove" onclick="remove(${index});">remove</button>
+              </div>
+              <hr>
+             </div>`;
+    });
   }
-};
+}
 
-removeBook();
+let collection = JSON.parse(localStorage.getItem('book'));
 
+if (collection === null) {
+  collection = [];
+}
+const bookArr = new Book(collection);
 addBtn.addEventListener('click', (e) => {
   e.preventDefault();
-  const newData = {
-    name: inputTitle.value,
-    author: inputAuthor.value,
-  };
-  collection.push(newData);
-  saveData();
-  displayData();
+  const book1 = { name: inputTitle.value, author: inputAuthor.value };
+  bookArr.saveData(book1);
+  bookArr.getData();
```

Comment:
- [ ] I would recommend adding validation to the name and author. Currently, I am able to create a book with empty details.

![Screenshot from 2021-10-20 20-20-47](https://user-images.githubusercontent.com/22434660/138117127-a71c3afb-98a5-4415-b292-447f9c59964e.png)

Reply:
This issue is well addressed. I also recommend you as an optional comment, that you empty the input fields after submission. :pray: :+1: :clap: